tests/unittest/bindEvict/main_test.go

```diff
@@ -100,38 +100,49 @@ func sleepyQ(conn *sql.Conn, delayRow int) error {
 	return nil
 }
 
-func fastAndSlowBinds() error {
+var normCliErr error
+
+func NormCliErr() error {
+	if normCliErr == nil {
+		normCliErr = fmt.Errorf("normal client got error")
+	}
+	return normCliErr
+}
+
+func partialBadLoad(fracBad float64) error {
 	db, err := sql.Open("hera", "127.0.0.1:31002")
 	if err != nil {
 		fmt.Printf("Error db %s\n", err.Error())
 		return err
 	}
-	db.SetConnMaxLifetime(22 * time.Second)
+	db.SetConnMaxLifetime(111 * time.Second)
 	db.SetMaxIdleConns(0)
 	db.SetMaxOpenConns(22111)
 	defer db.Close()
 
 	// client threads of slow queries
 	var stop2 int
+	var stop3 int
 	var badCliErr string
-	mkClients(1+int(max_conn*1.6), &stop2, 29001111, "badClient", &badCliErr, db)
+	var cliErr string
+	numBad := int(max_conn * fracBad)
+	numNorm := int(max_conn*2.1) + 1 - numBad
+	fmt.Printf("spawning clients bad%d norm%d\n", numBad, numNorm)
+	mkClients(numBad, &stop2, 29001111, "badClient", &badCliErr, db)
+	mkClients(numNorm, &stop3, 100, "normClient", &cliErr, db) // bind value is short, so bindevict won't trigger
 	time.Sleep(3100 * time.Millisecond)
-	/* if (testutil.RegexCountFile("BIND_THROTTLE", "cal.log") == 0) {
-		return fmt.Errorf("BIND_THROTTLE was not triggered")
-	}
-	if (testutil.RegexCountFile("BIND_EVICT", "cal.log") == 0) {
-		return fmt.Errorf("BIND_EVICT was not triggered")
-	} // */
+	//time.Sleep(33100 * time.Millisecond)
 
 	// start normal clients after initial backlog timeouts
-	var normCliErrStr string
 	var stop int
+	var normCliErrStr string
 	mkClients(1, &stop, 21001111, "n client", &normCliErrStr, db)
 	time.Sleep(1100 * time.Millisecond)
 
 	// if we throttle down or stop, it restores
 	stop2 = 1 // stop bad clients
-	lib.GetConfig().BindEvictionDecrPerSec = 11500.1
+	stop3 = 1
+	lib.GetConfig().BindEvictionDecrPerSec = 11333.1
```

Comment:
I see the only difference between this test and the one below (tests/unittest/bindThrottle/main_test.go) is the BindEvictionDecrPerSec. Can we have another test inside here rather than duplicating the test?

Reply:
there is slight difference in bindEvict test, here we are overriding the BindEvictionDecrPerSec value but in bindThrottle test we are not overriding BindEvictionDecrPerSec value.